src/bootstrap/bin/rustc.rs

```diff
@@ -106,6 +106,10 @@ fn main() {
         {
             cmd.arg("-C").arg("panic=abort");
         }
+
+        if env::var("RUSTC_TLS_MODEL_INITIAL_EXEC").is_ok() {
```

Comment:
Once the try build is complete please add a comment explaining that this must not be set for proc macros and that this is the reason it is guarded behind `target.is_some()`. Don't push before the try build is complete though as that may abort the try build.